### PR TITLE
Archive layout 

### PR DESCRIPTION
**Pull request by JayTokBox**

### CODE REVIEW

File: lib/opentok/client.rb [hunk 1]
```diff
@@ -164,6 +164,28 @@ def delete_archive(archive_id)
...
+      opts.extend(HashExtensions)
+      response = self.class.put("/v2/project/#{@api_key}/archive/#{archive_id}/layout", {
+          :body => opts.camelize_keys!.to_json,
+          :headers => generate_headers("Content-Type" => "application/json")
+      })
+      case response.code
+      when 200
+        response
+      when 400
+        raise OpenTokArchiveError, "Setting the layout failed.The request was invalid or maybe invalid layout options were given."
```
Comment (normanargueta) [inline lib/opentok/client.rb]: @msach22 should we use the same error messages that are described in the documentation? https://tokbox.com/developer/rest/#change_composed_archive_layout

**Comment by normanargueta** [comment]
@JayTokBox please check at the inline comments, thanks!

File: lib/opentok/archive.rb [hunk 2]
```diff
@@ -95,6 +95,16 @@ def delete
       @json = @interface.delete_by_id @json['id']
     end
 
+    # Layouts an OpenTok archive.
+    #
+    # You can only delete an archive which has a status of "available" or "uploaded". Deleting an
```
[the remark this answers is not in the document]
Comment (JayTokBox) [reply lib/opentok/archive.rb]: long live copy paste

File: lib/opentok/client.rb (hunk 1 above)
Comment (msach22) [reply lib/opentok/client.rb]: Yes, let's use the response that's the same as the API server

Comment (msach22) [reply lib/opentok/client.rb]: I think @JayTokBox is following the error handling that the SDK has been following so I think it's okay. We should clean up all SDKs to simply forward the REST API response.

Comment (normanargueta) [reply lib/opentok/client.rb]: Ok, sounds good @msach22 , so I will merge this PR after the last check

**Pull request merged**